dsn.go

```diff
@@ -26,32 +28,41 @@ func ParseDSN(dsn string) (*Config, error) {
 	}
 	id, key, endpointURL := sub[1], sub[2], sub[3]
 
-	query := make(map[string]string)
-	for _, s := range strings.Split(sub[4], "&") {
-		pair := reQuery.FindStringSubmatch(s)
-		if len(pair) != 3 {
-			return nil, fmt.Errorf("dsn %s doesn't match access_id:access_key@url?curr_project=project&scheme=http|https", dsn)
-		}
-		if pair[1] != "scheme" && pair[1] != currentProject {
-			return nil, fmt.Errorf("dsn %s 's query is neither scheme or %s", dsn, currentProject)
-		}
-		query[pair[1]] = pair[2]
+	var schemeArgs []string
+	var currProjArgs []string
+	var ok bool
+	queryHints := make(map[string]string)
+
+	querys, err := url.ParseQuery(sub[4])
+	if err != nil {
+		return nil, err
+	}
+
+	if schemeArgs, ok = querys["scheme"]; !ok || len(schemeArgs) != 1 {
+		return nil, fmt.Errorf("dsn %s should have one scheme argument", dsn)
 	}
-	if _, ok := query[currentProject]; !ok {
-		return nil, fmt.Errorf("dsn %s doesn't have curr_project", dsn)
+	if currProjArgs, ok = querys[currentProject]; !ok || len(currProjArgs) != 1 {
+		return nil, fmt.Errorf("dsn %s should have one current_project argument", dsn)
 	}
-	if _, ok := query["scheme"]; !ok {
-		return nil, fmt.Errorf("dsn %s doesn't have scheme", dsn)
+
+	for k, v := range querys {
+		// The query args such as hints_odps.sql.mapper.split_size=16
+		// would be converted to the maxcompute query hints: {"odps.sql.mapper.split_size": "16"}
+		if strings.HasPrefix(k, "hints_") {
```

Comment:
How about using `hint` as a prefix?
Because the `DSN` maybe looks like `gomaxcompute://user:passwd@host/project?hint_odps.sql.mapper.split_size=16&hint_odps.sql.other.key=value`, hints separated by a `&`.

Reply:
You're right, thanks and will fix that.